proto/src/determined/api/v1/experiment.proto

```diff
@@ -246,3 +246,37 @@ message CreateExperimentResponse {
   // The created experiment config.
   google.protobuf.Struct config = 2;
 }
+
+// Request for the set of metrics recorded by an experiment.
+message MetricNamesRequest {
+  // The id of the experiment.
+  int32 experiment_id = 1;
+}
+
+// Response to MetricNamesRequest.
+message MetricNamesResponse {
+  // The name of the searcher metric
+  string searcher_metric = 1;
+  // List of training metric names.
+  repeated string training_metrics = 2;
+  // List of validation metric names.
+  repeated string validation_metrics = 3;
+}
+
+// Request the milestones (in batches processed) at which a metric is recorded
+// by an experiment.
+message MetricBatchesRequest {
+  // The id of the experiment.
+  int32 experiment_id = 1;
+  // A training metric name (this or validation_metric required, not both).
+  string training_metric = 2;
+  // A validation metric name (this or training_metric required, not both).
+  string validation_metric = 3;
```

Comment:
non-blocking: feelings about using a `MetricType` enum out at this level instead of these two options, so then the invalid input isn't even possible, rather than all this validation internally checking the size of strings?

Reply:
Yeah that works for me. @hkang1 does that sound good to you (as the future consumer of this)?

Reply:
(Confirmed with Caleb offline)